File: python/28-find_nemo_timed.py
from datetime import datetime


def find_nemo(array):
    t1 = datetime.now()
    for item in array:
        if item == 'nemo':
            print('found NEMO')
    t2 = datetime.now()
    print(f'Call to find nemo took {t2 - t1} milliseconds')


nemo = ['nemo']
everyone = ['dory', 'bruce', 'marlin', "nemo", 'gill', 'bloat', 'nigel', 'squirt', 'darla', 'hank']
large = ['test' for _ in range(100000)]

find_nemo(nemo)
find_nemo(everyone)
find_nemo(large)

File: javascript/28-find_nemo_timed.js
const {performance} = require('perf_hooks');
const nemo = ['nemo'];
const everyone = ['dory', 'bruce', 'marlin', "nemo", 'gill', 'bloat', 'nigel', 'squirt', 'darla', 'hank'];
const large = new Array(1000000).fill('test');


function findNemo (array) {
    let t0 = performance.now();
    for (let i = 0; i < array.length; i++) {
        if (array[i] === 'nemo') {
            console.log('Found NEMO')
        }
    }
    let t1 = performance.now();
    console.log('Call to find Nemo took ' + (t1-t0) + 'milliseconds');
}

findNemo(nemo);
findNemo(everyone);
findNemo(large);
